File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').trim().split('\n');

const [N, M] = input[0].split(" ").map(Number);
let basket = Array.from({ length: N }, (_, i) => i + 1);

for (let n = 0; n < M; n++) {
  const [i, j] = input[n + 1].split(" ").map(Number);
  const target = [...basket].splice(i - 1, j - (i - 1));
  basket.splice(i - 1, j - (i - 1), ...target.reverse());
}

console.log(basket.join(" "));

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.swift
let cond: [Int] = readLine()!.split(separator: " ").map { Int($0)! }
let (n, m): (Int, Int) = (cond[0], cond[1])
var basket: [String] = Array(1 ... n).map { String($0) }

for _ in 0 ..< m {
  let input = readLine()!.split(separator: " ").map { Int($0)! }
  let range = input[0] - 1 ... input[1] - 1
  
  var target = basket[range]
  target.reverse()
  
  basket.replaceSubrange(range, with: target)
}

print(basket.joined(separator: " "))